galley/pkg/config/source/kube/apiserver/status/controller.go

```diff
@@ -141,6 +141,10 @@ func (c *ControllerImpl) Report(messages diag.Messages) {
 		msgs.Add(origin, m)
 	}
 
+	if c.state == nil {
```

Comment:
Same comment as before.  Putting the if right before the use makes this more reliable but don't we need to guard against it going nil between this line and `applyMessages()`?

Reply:
This function is already only called while the `mu` lock is held, so it cannot be set to nil between these lines.

Reply:
will add rlock here